pkg/vmprovider/providers/vsphere2/network/network.go

```diff
@@ -0,0 +1,631 @@
+
+		for _, addr := range interfaceSpec.Addresses {
+			ip, _, err := net.ParseCIDR(addr)
+			if err != nil {
+				continue
+			}
+
+			ipConfig := NetworkInterfaceIPConfig{
+				IPCIDR: addr,
+				IsIPv4: ip.To4() != nil,
+			}
+
+			if ipConfig.IsIPv4 {
+				dhcp4 = false
+				ipConfig.Gateway = interfaceSpec.Gateway4
+			} else {
+				dhcp6 = false
+				ipConfig.Gateway = interfaceSpec.Gateway6
+			}
+
+			result.IPConfigs = append(result.IPConfigs, ipConfig)
+		}
+	}
+
+	result.Name = interfaceSpec.Name
+	result.DHCP4 = dhcp4
+	result.DHCP6 = dhcp6
+	result.Nameservers = interfaceSpec.Nameservers
+	result.SearchDomains = interfaceSpec.SearchDomains
+
+	if interfaceSpec.MTU != nil {
+		result.MTU = *interfaceSpec.MTU
+	}
+	for _, route := range interfaceSpec.Routes {
+		result.Routes = append(result.Routes, NetworkInterfaceRoute{To: route.To, Via: route.Via, Metric: route.Metric})
+	}
+}
+
+func createNamedNetworkInterface(
+	vmCtx context.VirtualMachineContextA2,
+	finder *find.Finder,
+	interfaceSpec *vmopv1.VirtualMachineNetworkInterfaceSpec) (*NetworkInterfaceResult, error) {
+
+	if interfaceSpec.Network.Kind != "" || interfaceSpec.Network.APIVersion != "" {
+		return nil, fmt.Errorf("network TypeMeta not supported for name network: %v", interfaceSpec.Network.TypeMeta)
+	}
+
+	networkName := interfaceSpec.Network.Name
+	if networkName == "" {
+		return nil, fmt.Errorf("network name is required")
+	}
+
+	backing, err := finder.Network(vmCtx, networkName)
+	if err != nil {
+		return nil, fmt.Errorf("unable to find named network %q: %w", networkName, err)
+	}
+
+	return &NetworkInterfaceResult{
+		NetworkID: networkName,
+		Backing:   backing,
+	}, nil
+}
+
+// NetOPCRName returns the name to be used for the NetOP NetworkInterface CR.
+func NetOPCRName(vmName, networkName, interfaceName string, isV1A1 bool) string {
+	var name string
+
+	if isV1A1 {
+		// Old naming convention: each network can really only have 1 NIC.
+		if networkName != "" {
+			name = fmt.Sprintf("%s-%s", networkName, vmName)
+		} else {
+			name = vmName
+		}
+	} else {
+		if networkName != "" {
+			name = fmt.Sprintf("%s-%s-%s", vmName, networkName, interfaceName)
+		} else {
+			name = fmt.Sprintf("%s-%s", vmName, interfaceName)
+		}
+	}
+
+	return name
+}
+
+func createNetOPNetworkInterface(
+	vmCtx context.VirtualMachineContextA2,
+	client ctrlruntime.Client,
+	vimClient *vim25.Client,
+	interfaceSpec *vmopv1.VirtualMachineNetworkInterfaceSpec) (*NetworkInterfaceResult, error) {
+
+	if kind := interfaceSpec.Network.Kind; kind != "" && kind != "Network" {
+		return nil, fmt.Errorf("network kind %q is not supported for VDS", kind)
+	}
+
+	// If empty, NetOP will try to select a namespace default.
+	networkName := interfaceSpec.Network.Name
+
+	netIf := &netopv1alpha1.NetworkInterface{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      NetOPCRName(vmCtx.VM.Name, networkName, interfaceSpec.Name, false),
```

Comment:
Should we pass in the FSS val for v1a2 here to the last argument?. or Is it referring to a different v1a1.